configure: automatically detect pkg-config to use, like CC

averyterrel opened the pull request:
Currently, when cross compiling, available libraries are incorrectly detected. This fixes that, and allows changing pkg-config implementations.

classabbyamp commented on configure:
```diff
@@ -189,6 +189,27 @@ else
+		[ -n "$PKG_CONFIG" ] && break
+	done
+	if [ -z "$PKG_CONFIG" ]; then
+		echo "not found, exiting."
```
```suggestion
		echo
		echo "no suitable pkg-config tool found - aborting" >&2
```
to match the `$CC` check

averyterrel committed:
configure: automatically detect pkg-config to use, like CC

Currently, when cross compiling, available libraries are incorrectly detected.
This fixes that, and allows changing pkg-config implementations.
